src/main/java/bio/terra/service/filedata/azure/AzureSynapsePdao.java

```diff
@@ -0,0 +1,310 @@
+  }
+
+  private String buildWithStatement(FormatEnum formatType, DatasetTable datasetTable) {
+    switch (formatType) {
+      case CSV:
+        List<Column> columns = datasetTable.getColumns();
+        return String.join(
+            ",\n      ",
+            columns.stream()
+                .map(c -> c.getName() + " " + translateTypeToDdl(c.getType(), c.isArrayOf()))
+                .collect(Collectors.toList()));
+      case JSON:
+        return "doc nvarchar(max)";
+      default:
+        throw new EnumConstantNotPresentException(FormatEnum.class, formatType.name());
+    }
+  }
+
+  private String addArguments(FormatEnum formatType, int csvSkipLeadingRows) {
+    switch (formatType) {
+      case CSV:
+        return "PARSER_VERSION = '"
+            + PARSER_VERSION
+            + "'"
+            + ",\n                                "
+            + "FIRSTROW = "
+            + csvSkipLeadingRows;
+      case JSON:
+        return "fieldterminator ='0x0b'"
+            + ",\n                                "
+            + "fieldquote = '0x0b'";
+      default:
+        throw new EnumConstantNotPresentException(FormatEnum.class, formatType.name());
+    }
+  }
+
+  public void dropTables(List<String> tableNames) {
+    tableNames.stream()
+        .forEach(
+            tableName -> {
+              try {
+                executeSynapseQuery("DROP EXTERNAL TABLE [" + tableName + "];");
+              } catch (Exception ex) {
+                logger.warn("Unable to clean up table {}, ex: {}", tableName, ex.getMessage());
+              }
+            });
+  }
+
+  public void dropDataSources(List<String> dataSourceNames) {
+    dataSourceNames.stream()
+        .forEach(
+            dataSource -> {
+              try {
+                executeSynapseQuery("DROP EXTERNAL DATA SOURCE [" + dataSource + "];");
+              } catch (Exception ex) {
+                logger.warn(
+                    "Unable to clean up the external data source {}, ex: {}",
+                    dataSource,
+                    ex.getMessage());
+              }
+            });
+  }
+
+  public void dropScopedCredentials(List<String> credentialNames) {
+    credentialNames.stream()
+        .forEach(
+            credential -> {
+              try {
+                executeSynapseQuery("DROP DATABASE SCOPED CREDENTIAL [" + credential + "];");
+              } catch (Exception ex) {
+                logger.warn(
+                    "Unable to clean up scoped credential {}, ex: {}", credential, ex.getMessage());
+              }
+            });
+  }
+
+  public boolean executeSynapseQuery(String query) throws SQLException {
+    SQLServerDataSource ds = getDatasource();
+    try {
+      Connection connection = ds.getConnection();
+      Statement statement = connection.createStatement();
+      return statement.execute(query);
+    } catch (SQLException throwables) {
+      throw new SQLException("Synapse Query Failed.", throwables);
+    }
+  }
```

Comment:
@nmalfroy - spotbugs is complaining that we are not cleaning this up. Is there a reason for wanting to combine this into a single block? I think I would fix the spotbugs by adding try/finally statements for both the connection and statement.

Reply:
Note that you can do both in a single try
```
try (Connection connection = ds.getConnection();
      Statement statement = connection.createStatement()) {
  ...
} catch (SQLException e) {
  ...
}
```
Also I would use `e` instead of `throwables`

Reply:
Right, that had been my original suggestion (sorry it if wasn't clear).  Putting both in that try-with-resources block should make both of those close

Reply:
Ahhh gotcha, yeah - that was the original suggestion, I just misunderstood the suggested change.